Sources/Web3Core/EthereumABI/ABIParsing.swift

```diff
@@ -7,16 +7,41 @@ import Foundation
 
 extension ABI {
 
-    public enum ParsingError: Swift.Error {
+    public enum ParsingError: LocalizedError {
         case invalidJsonFile
-        case elementTypeInvalid
+        case elementTypeInvalid(_ desc: String? = nil)
         case elementNameInvalid
         case functionInputInvalid
         case functionOutputInvalid
         case eventInputInvalid
         case parameterTypeInvalid
         case parameterTypeNotFound
         case abiInvalid
```

Comment:
I've decided not to add `(_ desc: String? = nil)` too all error types as it would trigger more changes across the library.
It's better to do that updated in a separate PR.